src/prefect/server/schemas/core.py

```diff
@@ -654,6 +663,10 @@ class ConcurrencyLimitV2(ORMBaseModel):
     avg_slot_occupancy_seconds: float = Field(
         default=2.0, description="The average amount of time a slot is occupied."
     )
+    used_for: Optional[InternalConcurrencyLimitUsedFor] = Field(
```

Comment:
Hmmm having this field be something clients need to be responsible for feels off if it's purpose is to signify a "private" limit -- were other design alternatives considered for this that I could compare against?